curl-helper.c

```diff
@@ -4658,7 +4633,9 @@ value caml_curl_multi_init(value unit)
   }
 
   multi->handle = h;
-  multi->values = caml_alloc(multi_values_total, 0);
+  multi->values = caml_alloc_small(multi_values_total, 0);
+  for (int i = 0; i < multi_values_total; i++)
+    Store_field(multi->values, i, Val_unit);
```

Comment:
Can you say a word about this change? It looks buggy: one should not use `caml_modify` (or its avatar, `Store_field`) with blocks allocated with `caml_alloc_small` which are not initialized and thus contain garbage.

Taking a step back, what was wrong with the original code?

Reply:
My goal here is to initialize multi->values with a known sentinel value (Val_unit) so that I can later check if the callback has been set when within caml_curl_multi_add_handle. If there's a better way to do that, that'd be fine as well.

You're right that I should use Field() and not Store_field() with caml_alloc_small. I had thought thtat _small was preferable when allocating a small amount of memory, which we are, but caml_alloc is probably also fine.

I changed this to Field() but let me know if you'd like something different.

Reply:
Nothing needs to be changed: `caml_alloc` already initializes every field of the block with `Val_unit`.

Reply:
Done. I didn't know about the Val_unit initialization, I'll go and read the source for these functions later.